### ISSUE TITLE
Second attempt: restrict using unused variables

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#376]
# Rule request

## Thesis

We need to restrict situations like:

```python
def test():
    _text = 'print me!'
    print(_text)
```

What variables should be checked?

1. function / module scoped local variables

What variables should not be checked?

1. class attributes, `_` indicates access rights, not used / unused
2. arguments, since adding extra `_` to argument names is changing function's signature
3. module level constants, `_` indicates access rights, not used / unused

This time we are limiting the scope of this task very carefully. 
What variables are considered unused? Local variables starting with `_`, or just `_`.

## Reasoning

We should not allow adding extra prefixes to variable names just because a user wants to.
This process has it meaning and we should check that it is working correctly.

**Comment by sobolevn** [comment]
Related: https://github.com/wemake-services/wemake-python-styleguide/issues/317
